server/graphql/v2/query/collection/TransactionsCollectionQuery.ts

```diff
@@ -109,46 +114,71 @@ const TransactionsCollectionQuery = {
         reference => reference && fetchAccountWithReference(reference, fetchAccountParams),
       ),
     );
+
     if (fromAccount) {
-      let fromCollectiveCondition = fromAccount.id;
+      const fromAccountCondition = [fromAccount.id];
+
+      if (args.includeChildrenTransactions) {
+        const childIds = await fromAccount.getChildren().then(children => children.map(child => child.id));
+        fromAccountCondition.push(...childIds);
+      }
+
+      // When users are admins, also fetch their incognito contributions
       if (
         args.includeIncognitoTransactions &&
         req.remoteUser?.isAdminOfCollective(fromAccount) &&
         req.remoteUser.CollectiveId === fromAccount.id
       ) {
-        const incognitoProfile = await req.remoteUser.getIncognitoProfile();
-        if (incognitoProfile) {
-          fromCollectiveCondition = { [Op.or]: [fromAccount.id, incognitoProfile.id] };
+        const fromAccountUser = await fromAccount.getUser();
+        if (fromAccountUser) {
+          const incognitoProfile = await fromAccountUser.getIncognitoProfile();
+          if (incognitoProfile) {
+            fromAccountCondition.push(incognitoProfile.id);
+          }
```

Comment:
Refactoring opportunity to fix what seems a wrong implementation of incognito filter.

Reply:
Previous implementation was actually "ok" given conditions to enter the block.